fpdf/fonts.py

```diff
@@ -267,6 +271,34 @@ def __init__(self, fpdf, font_file_path, fontkey, style):
         )
 
         self.scale = 1000 / self.ttfont["head"].unitsPerEm
+
+        # check if the font is a TrueType and missing a .notdef glyph
+        # if it is missing, provide a fallback glyph
+        if "glyf" in self.ttfont and ".notdef" not in self.ttfont["glyf"]:
+            LOGGER.warning(
+                (
+                    "TrueType Font '%s' is missing the '.notdef' glyph. "
+                    "Fallback glyph will be provided."
+                ),
+                self.fontkey,
+            )
+            # draw a diagonal cross .notdef glyph
+            pen = TTGlyphPen(self.ttfont["glyf"])
+            pen.moveTo((0, 0))
+            pen.lineTo((600, 0))
+            pen.lineTo((600, 600))
+            pen.lineTo((0, 600))
+            pen.closePath()
+            pen.moveTo((0, 0))
+            pen.lineTo((600, 600))
+            pen.closePath()
+            pen.moveTo((600, 0))
+            pen.lineTo((0, 600))
+            pen.closePath()
+
+            self.ttfont["glyf"][".notdef"] = pen.glyph()
+            self.ttfont["hmtx"][".notdef"] = (600, 0)
+
```

Comment:
I believe having the fixed 0,0 to 600,600 would look weird on different fonts.
Maybe having your coordinates set to (ttfont['head'].xMin, ttfont['head'].yMax), (ttfont['head'].xMax, ttfont['head'].yMax) would generate a more consistent notdef outline.